### PR TITLE
feat: Created Upstash Redis Chat Memory for Edge and Serverless environments

### PR DESCRIPTION
**Pull request by henriquegdantas**
**Hello!**

This is my first contribution, so if there's something that needs to be changed, I'm open to suggestions.

It's based on the Redis Chat Memory, but the upstash client uses HTTP instead of a TCP connection, which allows us to use this on **Edge and Serverless environments**.

This was an important feature to me so I hope it helps other people.
I added docs and tests, too.

Thanks for the awesome work on langchain.

### CODE REVIEW

File: langchain/.env.example [hunk 1]
```diff
@@ -30,3 +30,5 @@ MYSCALE_PORT=ADD_YOURS_HERE
 MYSCALE_USERNAME=ADD_YOURS_HERE
 MYSCALE_PASSWORD=ADD_YOURS_HERE
 REDIS_URL=ADD_YOURS_HERE
+UPSTASH_REDIS_REST_URL=https://ADD_YOURS_HERE.upstash.io
+REDIS_UPUPSTASH_REDIS_REST_TOKENSTASH_TOKEN=ADD_YOURS_HERE
```
Comment (jacoblee93) [inline langchain/.env.example:34]: Bad find and replace looks like?

File: langchain/scripts/create-entrypoints.js [hunk 2]
```diff
@@ -129,6 +129,7 @@ const entrypoints = {
   "stores/file/node": "stores/file/node",
   "stores/message/dynamodb": "stores/message/dynamodb",
   "stores/message/redis": "stores/message/redis",
+  "stores/message/redis_upstash": "stores/message/redis_upstash",
```
Comment (jacoblee93) [inline langchain/scripts/create-entrypoints.js:132]: Can we reverse the order of all these names to `upstash_redis` to mirror the company's name?

**Review by jacoblee93 — CHANGES_REQUESTED** [review]

**Comment by jacoblee93** [comment]
Looks good, just a few small comments!

**Comment by jacoblee93** [comment]
Some polish and `s/upstash_redis/redis_upstash` to better match Upstash's language. 

Thank you very much, this will help a lot of people in serverless environments!

**Review by jacoblee93 — APPROVED** [review]

**Comment by henriquegdantas** [comment]
> Some polish and `s/upstash_redis/redis_upstash` to better match Upstash's language.
> 
> Thank you very much, this will help a lot of people in serverless environments!

Sorry for not replying earlier. Had a super busy week.
Thanks for the fixes! Glad I could help.

**Pull request merged**